celery/canvas.py

```diff
@@ -1072,11 +1072,8 @@ def skew(self, start=1.0, stop=None, step=1.0):
     def apply_async(self, args=None, kwargs=None, add_to_parent=True,
                     producer=None, link=None, link_error=None, **options):
         args = args if args else ()
-        if link is not None:
-            raise TypeError('Cannot add link to group: use a chord')
-        if link_error is not None:
-            raise TypeError(
-                'Cannot add link to group: do that on individual tasks')
```

Comment:
Why change the exception message?

Reply:
No real reason - I can change the text back. The actual change here is to falsey checks rather than `None` checks since linking an empty iterable of tasks does nothing anyway so we can probably allow it. That said, doing so could allow weird behaviours to go unnoticed or introduce "sometimes it breaks" conditions (when there is a link/link_error) which could be annoying.

Happy to pull this change out of the diff. It's incidental and was only added because I had an unconditional `self.request.callbacks or []` in `task.py` before I decided that wasn't very nice.

Reply:
I'd rather not break our public API (and exceptions are public API).
If you'd like, we can raise something more specific that inherits from a TypeError.

Reply:
Sure thing, I'll just strip it out since it's not a particularly useful change. Will resolve once I push a new diff.